UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -237,5 +237,40 @@ public async Task<byte[]> DownloadFSSZipFileAsync(string batchId, string fileNam
             }
 
         }
+
+        public async Task<List<ShowFilesResponseModel>> GetCumulativeBatchFiles(string correlationId)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.GetCumulativeFilesResponseStarted.ToEventId(), "Maritime safety information request to get cumulative NM files response started with _X-Correlation-ID:{correlationId}", correlationId);
+
+                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(correlationId);
+
+                string searchText = $" and $batch(Frequency) eq 'cumulative'";
+
+                IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+
+                IResult<BatchSearchResponse> result = await _fileShareService.FSSBatchSearchAsync(searchText, accessToken, correlationId, fileShareApiClient);
+
+                BatchSearchResponse SearchResult = result.Data;
+
+                if (SearchResult != null && SearchResult.Entries.Count > 0)
+                {
+                    List<ShowFilesResponseModel> showFilesResponseModel = NMHelper.ListFilesResponseCumulative(SearchResult.Entries).OrderByDescending(e => NMHelper.GetYearAndTypeFromFilenName(e.FileDescription)).ToList();
```

Comment:
As per changes to helper method, this will be changed as below.
```suggestion
                    List<ShowFilesResponseModel> showFilesResponseModel = NMHelper.ListFilesResponseCumulative(SearchResult.Entries).ToList();
```